lib/promscrape/discovery/ec2/api.go

```diff
@@ -135,19 +254,79 @@ func getMetadataByPath(apiPath string) ([]byte, error) {
 	return readResponseBody(resp, apiURL)
 }
 
-func getAPIResponse(cfg *apiConfig, action, nextPageToken string) ([]byte, error) {
+// refreshRoleArnCredentials - retrieves credentials data from aws sts
+// using role_arn
+// updates apiConfig
+func (a *apiConfig) refreshRoleArnCredentials() error {
+	apiEndpoint := buildAPIEndpoint(a.region, a.endpoint, "sts")
```

Comment:
Please move `buildAPIEndpoint()` call to the beginning of `getSTSAPIResponse()` function, since the returned endpoint isn't used outside this function.